oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1639,16 +1640,37 @@ public void removeGarbage(final VersionGCStats stats) {
                                     update.getId());
                             continue;
                         }
+                        NodeState traversedParent = null;
+                        NodeState traversedState = root;
+                        for (String name : oldDoc.getPath().elements()) {
+                            traversedParent = traversedState;
+                            traversedState = traversedState.getChildNode(name);
+                        }
+
                         NodeDocument newDoc = Collection.NODES.newDocument(ds);
                         oldDoc.deepCopy(newDoc);
                         UpdateUtils.applyChanges(newDoc, update);
-                        if (!verify(oldDoc, newDoc, update)) {
+                        // for the time being, verify both with classic and traversed
+                        if (!verifyViaTraversedState(traversedState, traversedParent,
```

Comment:
Should we maybe add a debug line here? We may have interest in detecting these cases. Wdyt?

Reply:
+1, added https://github.com/apache/jackrabbit-oak/pull/1377/commits/5b668577a5712f48fa569d998e99ae3fc302c362